math/src/fft/fft_inputs.rs

```diff
@@ -0,0 +1,81 @@
+use super::StarkField;
+use crate::FieldElement;
+
+// FFTINPUTS TRAIT
+// ================================================================================================
+
+/// Defines the interface that must be implemented by the input to fft_in_place method.
+pub trait FftInputs<B: StarkField> {
```

Comment:
Looks good! One thing I wonder if we should put `permute()` as a function on this trait. It would be an implemented  function here. Something like:
```Rust
fn permute(&mut self) {
    let n = self.len();
    for i in 0..n {
        let j = permute_index(n, i);
        if j > i {
            self.swap(i, j);
        }
    }
}
```

Reply:
Or, maybe we could put `fft_in_place()` here as well. Something like:
```Rust
fn fft_in_place(&mut self, twiddles: &[B]) {
    fft_in_place(self, twiddles, 1, 1, 0);
}
```
And then `fft_in_place()` function could be a helper function in this module.

Reply:
We have to also move the concurrent part of these method in `FftInput` crate. We can do it in the next PR.

Reply:
I was talking about these two methods in `concurrent.rs`.

- [permute](https://github.com/0xKanekiKen/winterfell/blob/add-fftinput-trait/math/src/fft/concurrent.rs#L95)
- [split_radix_fft](https://github.com/0xKanekiKen/winterfell/blob/add-fftinput-trait/math/src/fft/concurrent.rs#L123)

Reply:
Yep - makes sense!